src/connector/src/sink/iceberg/jni_catalog.rs

```diff
@@ -0,0 +1,262 @@
+            icelake::Error::new(
+                ErrorKind::Unexpected,
+                "Failed to attach current thread to jvm",
+            )
+            .set_source(e)
+        })?;
+
+        let mut env = self.jvm.get_env().map_err(|e| {
+            icelake::Error::new(ErrorKind::Unexpected, "Failed to get jni env").set_source(e)
+        })?;
+
+        let table_name_str = table_name.to_string();
+
+        let table_name_jstr = env.new_string(&table_name_str).unwrap();
+
+        let result_json = env
+            .call_method(
+                self.java_catalog.as_obj(),
+                "loadTable",
+                "(Ljava/lang/String;)Ljava/lang/String;",
+                &[(&table_name_jstr).into()],
+            )
+            .map_err(|e| {
+                icelake::Error::new(
+                    ErrorKind::Unexpected,
+                    "Failed to call loadTable in jni catalog",
+                )
+                .set_source(e)
+                .with_context("table_name", &table_name_str)
+            })?;
+
+        let rust_json_str = unsafe {
+            let jvalue = result_json.as_jni();
+            let jstring = JString::from_raw(jvalue.l);
+            let java_str = env.get_string(&jstring).map_err(|e| {
+                icelake::Error::new(
+                    ErrorKind::Unexpected,
+                    "Failed to get string from return value of loadTable method of jni catalog",
+                )
+                .set_source(e)
+            })?;
+
+            java_str
+                .to_str()
+                .map_err(|e| {
+                    icelake::Error::new(
+                        ErrorKind::Unexpected,
+                        "Failed to convert java string to rust utf8 string",
+                    )
+                    .set_source(e)
+                    .with_context("Java string bytes", format!("{:?}", java_str.to_bytes()))
+                })?
+                .to_string()
+        };
+
+        let resp: LoadTableResult = serde_json::from_str(&rust_json_str)?;
+
+        let metadata_location = resp.metadata_location.clone().ok_or_else(|| {
+            icelake::Error::new(
+                icelake::ErrorKind::IcebergFeatureUnsupported,
+                "Loading uncommitted table is not supported!",
+            )
+        })?;
+
+        tracing::info!("Table metadata location of {table_name} is {metadata_location}");
+
+        let table_metadata = resp.table_metadata()?;
+
+        let iceberg_io_args = IcebergTableIoArgs::builder_from_path(&table_metadata.location)?
+            .with_args(self.config.table_io_configs.iter())
+            .build()?;
+        let table_op = iceberg_io_args.create()?;
+
+        Ok(
+            Table::builder_from_catalog(table_op, self.clone(), table_metadata, table_name.clone())
+                .build()?,
+        )
+    }
+
+    async fn update_table(self: Arc<Self>, update_table: &UpdateTable) -> icelake::Result<Table> {
```

Comment:
This method has not been used. In which circumstances we will use this method?

Reply:
It's used in coordinator to commit updates to table.